### ISSUE TITLE
png resizing only working first time

### ISSUE THREAD

**Issue by tschiela** [lovell/sharp#1602]
node: 10.13.0
sharp: 0.21.3

I try to transform JPEG und PNG in an AWS hosted docker image. I have no problems with JPEG, but with PNG. Some rezises work great, some not. When I resize to mutch images in nearly the same time I got more / only errors. On my dev machine (less memory then production) i got mutch more errors. Is maybe the memory the cause of the problem? If so, how i can avoid it? 

```
Error: vipspng: libpng read error
vips2png: unable to write "/tmp/8axpnNmv3-original.png"
```

Dockerfile
```
FROM node:10.13.0-alpine

RUN mkdir -p /usr/src/app
RUN mkdir -p /usr/src/app/backend
RUN mkdir -p /usr/src/app/frontend

RUN apk add --no-cache \
    tzdata \
    git \
    python \
    build-base

ENV TZ Europe/Berlin

WORKDIR /usr/src/app

COPY backend /usr/src/app/backend
COPY package.json /usr/src/app
COPY frontend /usr/src/app/frontend

RUN npm config set unsafe-perm true
RUN npm install forever -g
RUN npm install

EXPOSE 80

CMD [ "forever", "backend/server.js", "-f"]
```

nodejs
```
sharp(srcBuffer)
      .jpeg({ quality: 75, force: false })
      .png({ compressionLevel: 9, force: false })
      .resize(sharpConf)
      .toFile(destPath);
```

I also tried debian as docker base image, no change at all.

**Comment by lovell** [comment]
Hello, the `vipspng: libpng read error` message suggests an input PNG image is corrupt/invalid. Are you able to provide an input PNG that consistently fails in this manner?

**Comment by AllSeeingEye** [comment]
I'm getting the same error, and it seems that libpng just cannot be used in parallel threads.

**Comment by lovell** [comment]
@AllSeeingEye Please can you open a new issue and provide complete, standalone code and sample image(s) that allow someone else to reproduce.
